sdk/servicebus/azure-messaging-servicebus/src/main/java/com/azure/messaging/servicebus/ServiceBusMessage.java

```diff
@@ -108,23 +109,109 @@ public ServiceBusMessage(BinaryData body) {
      * @param receivedMessage The received message to create new message from.
      *
      * @throws NullPointerException if {@code receivedMessage} is {@code null}.
+     * @throws UnsupportedOperationException if {@link AmqpMessageBodyType} is {@link AmqpMessageBodyType#SEQUENCE} or
+     * {@link AmqpMessageBodyType#VALUE}. See code sample above explaining how to check for {@link AmqpMessageBodyType}
+     * before calling this constructor.
+     * <p><strong>How to check for {@link AmqpMessageBodyType} before calling this constructor</strong></p>
+     * {@codesnippet com.azure.messaging.servicebus.ServiceBusMessage.copyServiceBusMessage}
+     * @throws IllegalStateException for invalid {@link AmqpMessageBodyType}.
+     * @see AmqpAnnotatedMessage
      */
     public ServiceBusMessage(ServiceBusReceivedMessage receivedMessage) {
         Objects.requireNonNull(receivedMessage, "'receivedMessage' cannot be null.");
 
-        this.amqpAnnotatedMessage = new AmqpAnnotatedMessage(receivedMessage.getAmqpAnnotatedMessage());
```

Comment:
Why was the copy-constructor removed from AmqpAnnotatedMessage? We did have the copy constructor earlier.

Reply:
I removed it for following reasons
1. Copy constructor is kind of Service bus needs and other library do not need it, for example Events hub.
2. dotnet also do not have it,  so to in sync with dotnet.